mmdet/datasets/dataset_wrappers.py

```diff
@@ -401,13 +401,16 @@ def __getitem__(self, idx):
                 continue
 
             if hasattr(transform, 'get_indexes'):
-                indexes = transform.get_indexes(self.dataset)
-                if not isinstance(indexes, collections.abc.Sequence):
-                    indexes = [indexes]
-                mix_results = [
-                    copy.deepcopy(self.dataset[index]) for index in indexes
-                ]
-                results['mix_results'] = mix_results
+                while True:
+                    indexes = transform.get_indexes(self.dataset)
+                    if not isinstance(indexes, collections.abc.Sequence):
+                        indexes = [indexes]
+                    mix_results = [
+                        copy.deepcopy(self.dataset[index]) for index in indexes
+                    ]
+                    if None not in mix_results:
```

Comment:
It seems that the transform will re-generate all samples even if there is only one None in mix_results, we may consider make it more efficient.